dll/win32/kernel32/kernel32_vista/GetFileInformationByHandleEx.c

```diff
@@ -3,68 +3,153 @@
-    case FileFullDirectoryRestartInfo:
-    case FileStorageInfo:
-    case FileAlignmentInfo:
-    case FileIdInfo:
-    case FileIdExtdDirectoryInfo:
-    case FileIdExtdDirectoryRestartInfo:
-        //FIXME( "%p, %u, %p, %u\n", handle, class, info, size );
-        SetLastError( ERROR_CALL_NOT_IMPLEMENTED );
-        return FALSE;
-
-    case FileBasicInfo:
-        status = NtQueryInformationFile( handle, &io, info, size, FileBasicInformation );
-        break;
-
-    case FileStandardInfo:
-        status = NtQueryInformationFile( handle, &io, info, size, FileStandardInformation );
-        break;
-
-    case FileNameInfo:
-        status = NtQueryInformationFile( handle, &io, info, size, FileNameInformation );
-        break;
-
-    case FileIdBothDirectoryRestartInfo:
-    case FileIdBothDirectoryInfo:
-        status = NtQueryDirectoryFile( handle, NULL, NULL, NULL, &io, info, size,
-                                       FileIdBothDirectoryInformation, FALSE, NULL,
-                                       (class == FileIdBothDirectoryRestartInfo) );
-        break;
-
-    case FileRenameInfo:
-    case FileDispositionInfo:
-    case FileAllocationInfo:
-    case FileIoPriorityHintInfo:
-    case FileEndOfFileInfo:
-    default:
-        SetLastError( ERROR_INVALID_PARAMETER );
-        return FALSE;
-    }
-
-    if (status != STATUS_SUCCESS)
-    {
-        SetLastError( RtlNtStatusToDosError( status ) );
-        return FALSE;
-    }
-    return TRUE;
+   union {
+      NTSTATUS Status;
+      PVOID Pointer;
+   } DUMMYUNIONNAME;
+
+   ULONG_PTR Information;
+} IO_STATUS_BLOCK, *PIO_STATUS_BLOCK;
+
+typedef NTSTATUS (CALLBACK *pfnNtQueryDirectoryFile)(
+   HANDLE                 FileHandle,
+   HANDLE                 Event,
+   PVOID                  ApcRoutine,
+   PVOID                  ApcContext,
+   PIO_STATUS_BLOCK       IoStatusBlock,
+   PVOID                  FileInformation,
+   ULONG                  Length,
+   FILE_INFORMATION_CLASS FileInformationClass,
+   BOOLEAN                ReturnSingleEntry,
+   WCHAR*                 FileName,
+   BOOLEAN                RestartScan);
+
+static pfnNtQueryDirectoryFile fnNtQueryDirectoryFile = nullptr;
+
+typedef NTSTATUS (CALLBACK *pfnNtQueryInformationFile)(
+   HANDLE                 FileHandle,
+   PIO_STATUS_BLOCK       IoStatusBlock,
+   PVOID                  FileInformation,
+   ULONG                  Length,
+   FILE_INFORMATION_CLASS FileInformationClass);
+
+static pfnNtQueryInformationFile fnNtQueryInformationFile = nullptr;
+
+static BOOL bNtdllInited = false;
+
+static BOOL WINAPI GetFileInformationByHandleEx(
+   _In_  HANDLE hFile,
+   _In_  FILE_INFO_BY_HANDLE_CLASS FileInformationClass,
+   _Out_writes_bytes_(dwBufferSize) LPVOID lpFileInformation,
+   _In_  DWORD dwBufferSize)
+{
+   FILE_INFORMATION_CLASS NtFileInformationClass;
+   DWORD cbMinBufferSize;
+   BOOLEAN RestartScan = false;
+
+   switch (FileInformationClass) // Purosham Sahanu InformationClass 
```

Comment:
Purosham Sahanu? This InformationClass is documented on MSDN...

Reply:
My search engine Ecosia didnt link me to MSDN , no complains